compiler/il/OMRNode.cpp

```diff
@@ -2472,15 +2556,45 @@ OMR::Node::computeIsCollectedReference()
             // only problem is: we might mark an aladd/aiadd on a temp whose value is null as
             // an internal pointer, thus creating a temp, pinning array and internal pointer
             // map for it.
-            return self() == curNode && curNode->getAddress() == 0;
+
+            // Preserve the existing logic which is as follows:
+            // true iff not under aladd and null,
+            // if non-null, always false.
+            // if under aladd, always false.
+
+            // non-null constant return false.
+            if (curNode->getAddress() != 0)
+               {
+               recordProcessedNodeResult(receiverNode, TR_no, processedNodesCollected, processedNodesNotCollected);
+               return TR_no;
+               }
+            else
+               {
+               // null constant under aladd, return false.
+               if (self() != curNode)
+                  {
+                  recordProcessedNodeResult(receiverNode, TR_no, processedNodesCollected, processedNodesNotCollected);
+                  return TR_no;
+                  }
+               else
+                  {
+                  // Null constant under ternary (i.e. we reached here via recursive calls), return maybe
+                  // to indicate need to check the other child.
+                  // Otherwise return yes for top level Null const nodes, to preserve previous logic.
+                  TR_YesNoMaybe result = (toplevel ? TR_yes : TR_maybe);
```

Comment:
this use of toplevel is now redundant I believe and the parameter should be eliminated.

Reply:
removed toplevel, it does look like redundant.